specification/ai/Azure.AI.Unified/datasets/models.tsp

```diff
@@ -0,0 +1,70 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-autorest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "../common/models.tsp";
+import "../main.tsp";
+import "@typespec/openapi";
+import "@typespec/versioning";
+
+using TypeSpec.OpenAPI;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Azure.AI.Unified;
+
+@doc("Enum to determine the type of data.")
+union DatasetType {
+  string,
+
+  @doc("URI file.")
+  uri_file: "uri_file",
+
+  @doc("URI folder.")
+  uri_folder: "uri_folder",
+}
+
+
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance"
+@doc("DatasetVersion Definition")
+@discriminator("datasetType")
+model DatasetVersion extends AssetBase {
+  @doc("[Required] Uri of the data. Example: https://go.microsoft.com/fwlink/?linkid=2202330")
+  @visibility("read", "create")
+  @minLength(1)
+  @pattern("[a-zA-Z0-9_]")
+  datasetUri: string;
+
+  @doc("Data type")
+  datasetType: DatasetType;
+}
+
+@doc("PagedDataVersionBase Definition")
+@pagedResult
+model PagedDatasetVersion {
```

Comment:
This is presumably the response used in a standard list operation - and if so, it should not be needed and we should use the resource list operation template.